cmd/earthly/main.go

```diff
@@ -1014,6 +1014,11 @@ func (app *earthlyApp) before(context *cli.Context) error {
 		return err
 	}
 
+	if app.strict == true && app.featureFlagOverrides != "" {
+		app.console.Warnf("EARTHLY_VERSION_FLAG_OVERRIDES (%s) will be ignored due to running in strict mode; they should be set via the VERSION Earthfile command instead\n", app.featureFlagOverrides)
+		app.featureFlagOverrides = ""
+	}
+
```

Comment:
A warning doesn't feel right here. I would either fail altogether if this is not allowed, or allow it under strict mode.

What's the argument against allowing it?